### PR TITLE
Delete post

### PR DESCRIPTION
**Pull request by mvn-tamnguyen4-dn**
## Description
Delete post after confirm
## Reference Requirements
- Product Backlog ID: PBI-11
- Task ID: 3

## Reference Isuees
None

## Migrations
NO

## Impacted Areas in Application
List general components of the application that this PR will affect:

-

## SQL query and `explain`
Begin Transaction

select * from `posts` where `posts`.`id` = '4' and `posts`.`deleted_at` is null limit 1
image

select * from `comments` where `comments`.`post_id` = '4' and `comments`.`post_id` is not null and `comments`.`deleted_at` is null
image

delete from `favorites` where `id` = '15'
image

update `comments` set `deleted_at` = '2018-01-30 08:49:34', `updated_at` = '2018-01-30 08:49:34' where `id` = '5'
image

update `comments` set `deleted_at` = '2018-01-30 08:49:34', `updated_at` = '2018-01-30 08:49:34' where `id` = '15'
image

update `posts` set `deleted_at` = '2018-01-30 08:49:34', `updated_at` = '2018-01-30 08:49:34' where `id` = '4'
image

Commit Transaction

### CODE REVIEW

File: app/Http/Controllers/Admin/PostController.php [hunk 1]
```diff
@@ -54,4 +54,18 @@ public function show($id)
...
+     *
+     * @param int $id id
+     *
+     * @return \Illuminate\Http\Response
+     */
+    public function destroy($id)
+    {
+        Post::find($id)->delete();
+        flash(__('post.delete_success'))->success();
+        return redirect()->route('posts.index');
```
Comment (at-anhngo) [inline app/Http/Controllers/Admin/PostController.php]: nếu lỡ không xóa được thì sao?, bỏ transaction vào

File: resources/views/backend/layouts/partials/js.blade.php [hunk 2]
```diff
@@ -31,4 +31,4 @@
 
 <script src="https://cdnjs.cloudflare.com/ajax/libs/bootstrap-datepicker/1.5.0/js/bootstrap-datepicker.js"></script>
 <script src="{{ asset('/js/delete_list.js') }}"></script>
-
+<script src="{{ asset('/js/delete_post.js') }}"></script>
```
Comment (at-anhngo) [inline resources/views/backend/layouts/partials/js.blade.php]: cái này chỉ sử dụng cho 1 chức năng thì bỏ vào view của chức năng đó thôi

**Review by ghost — CHANGES_REQUESTED** [review]
Check my comment

**Comment by at-anhngo** [comment]
LGTM

**Review by ghost — APPROVED** [review]

**Comment by at-duyho** [comment]
- explain mysql query in template, please, thank you!

File: app/Http/Controllers/Admin/PostController.php [hunk 3]
```diff
@@ -54,4 +55,31 @@ public function show($id)
...
+     *
+     * @param int $id id
+     *
+     * @return \Illuminate\Http\Response
+     */
+    public function destroy($id)
+    {
+        DB::beginTransaction();
+        try {
+            $post = Post::find($id);
```
Comment (at-duyho) [inline app/Http/Controllers/Admin/PostController.php]: - you should use findOrFail function and you do not check $post != null

**Review by ghost — CHANGES_REQUESTED** [review]
Check my comment

**Review by ghost — APPROVED** [review]
LGTM

**Pull request merged**